roachtest: npgsql failed

roachtest.npgsql [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/21353385?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/21353385?buildTab=artifacts#/npgsql) on master @ [d3d7be79ccdc6ee14ea4f75bcb241f326ed3e160](https://github.com/cockroachdb/cockroach/commits/d3d7be79ccdc6ee14ea4f75bcb241f326ed3e160):

Failed with:

```
(orm_helpers.go:225).summarizeFailed: 
Tests run on Cockroach v26.3.0-alpha.00000000-dev-d3d7be79ccdc6ee14ea4f75bcb241f326ed3e160
Tests run against npgsql v7.0.2
3432 Total Tests Run
2681 tests passed
751 tests failed
147 tests skipped
10 tests ignored
0 tests passed unexpectedly
2 tests failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: Npgsql.Tests.ReaderNewSchemaTests(Async).DefaultValue -  (unexpected)
--- FAIL: Npgsql.Tests.ReaderNewSchemaTests(Sync).DefaultValue -  (unexpected)
For a full summary look at the npgsql artifacts 
An updated blocklist (npgsqlBlocklist) is available in the artifacts' npgsql log
test artifacts and logs in: /artifacts/npgsql/run_1
```
Cluster Node to Ip Mapping:
| Node | Public IP | Private IP |
| --- | --- | --- |
| teamcity-21353385-1779256055-95-n1cpu4-0001 | 20.51.239.170 | 10.1.0.125 |

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>diskCount=0</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=leader</code>
 - <code>runtimeAssertionsBuild=false</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-21353385/npgsql/1779269830278/1779270164736)



See: [Datadog Logs](https://us5.datadoghq.com/logs?query=service%3Aroachtest+%40cluster%3Ateamcity-21353385-1779256055-95-n1cpu4&from_ts=1779269830278&to_ts=1779270164736&live=false&storage=flex_tier)

</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*npgsql.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-64098

roachtest.npgsql [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/21356543?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyAzureBazel/21356543?buildTab=artifacts#/npgsql) on master @ [248e519687c1d8129f5675a0fe0db9cf125c2b6d](https://github.com/cockroachdb/cockroach/commits/248e519687c1d8129f5675a0fe0db9cf125c2b6d):

Failed with:

```
(orm_helpers.go:225).summarizeFailed: 
Tests run on Cockroach v26.3.0-alpha.00000000-dev-248e519687c1d8129f5675a0fe0db9cf125c2b6d
Tests run against npgsql v7.0.2
3430 Total Tests Run
2680 tests passed
750 tests failed
149 tests skipped
10 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: Npgsql.Tests.ReaderNewSchemaTests(Async).DefaultValue -  (unexpected)
For a full summary look at the npgsql artifacts 
An updated blocklist (npgsqlBlocklist) is available in the artifacts' npgsql log
test artifacts and logs in: /artifacts/npgsql/run_1
```
Cluster Node to Ip Mapping:
| Node | Public IP | Private IP |
| --- | --- | --- |
| teamcity-21356543-1779428963-95-n1cpu4-0001 | 20.51.241.220 | 10.1.0.128 |

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=azure</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>diskCount=0</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=default</code>
 - <code>metamorphicWriteBuffering=true</code>
 - <code>runtimeAssertionsBuild=false</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-21356543/npgsql/1779440422585/1779440784592)



See: [Datadog Logs](https://us5.datadoghq.com/logs?query=service%3Aroachtest+%40cluster%3Ateamcity-21356543-1779428963-95-n1cpu4&from_ts=1779440422585&to_ts=1779440784592&live=false&storage=flex_tier)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*npgsql.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

roachtest.npgsql [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/21361128?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/21361128?buildTab=artifacts#/npgsql) on master @ [a081a00a50d84e8da201bbe55b898ec41b9b013b](https://github.com/cockroachdb/cockroach/commits/a081a00a50d84e8da201bbe55b898ec41b9b013b):

Failed with:

```
(orm_helpers.go:225).summarizeFailed: 
Tests run on Cockroach v26.3.0-alpha.00000000-dev-a081a00a50d84e8da201bbe55b898ec41b9b013b
Tests run against npgsql v7.0.2
3431 Total Tests Run
2680 tests passed
751 tests failed
148 tests skipped
10 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: Npgsql.Tests.ReaderNewSchemaTests(Async).DefaultValue -  (unexpected)
For a full summary look at the npgsql artifacts 
An updated blocklist (npgsqlBlocklist) is available in the artifacts' npgsql log
test artifacts and logs in: /artifacts/npgsql/run_1
```
Cluster Node to Ip Mapping:
| Node | Public IP | Private IP |
| --- | --- | --- |
| teamcity-21361128-1779863753-230-n1cpu4-0001 | 34.73.212.36 | 10.142.1.26 |

Parameters:
 - <code>arch=amd64</code>
 - <code>cloud=gce</code>
 - <code>coverageBuild=false</code>
 - <code>cpu=4</code>
 - <code>diskCount=0</code>
 - <code>encrypted=false</code>
 - <code>fs=ext4</code>
 - <code>localSSD=true</code>
 - <code>metamorphicLeases=expiration</code>
 - <code>metamorphicWriteBuffering=true</code>
 - <code>runtimeAssertionsBuild=false</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-21361128/npgsql/1779892002825/1779892401613)



See: [Datadog Logs](https://us5.datadoghq.com/logs?query=service%3Aroachtest+%40cluster%3Ateamcity-21361128-1779863753-230-n1cpu4&from_ts=1779892002825&to_ts=1779892401613&live=false&storage=flex_tier)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*npgsql.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/cmd/roachtest/tests/npgsql.go
+++ b/pkg/cmd/roachtest/tests/npgsql.go
@@ -58,6 +58,12 @@ func registerNpgsql(r registry.Registry) {
 			`GRANT admin TO npgsql_tests`,
 			`DROP DATABASE IF EXISTS npgsql_tests`,
 			`CREATE DATABASE npgsql_tests`,
+			// The npgsql test suite runs many tests in parallel. Concurrent DDL
+			// modify the database descriptor and cause concurrent transactions
+			// to fail with a retry error which npgsql does not retry. Retain
+			// the old descriptor versions to prevent the un-retried errors from
+			// percolating.
+			`SET CLUSTER SETTING sql.catalog.descriptor_lease.lock_old_versions.enabled = 'true'`,
 		} {
 			if _, err := db.ExecContext(ctx, cmd); err != nil {
 				t.Fatal(err)

